packages/node-core/src/indexer/postgresBasedDb.ts

```diff
@@ -0,0 +1,142 @@
+// Copyright 2020-2022 OnFinality Limited authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import {Sequelize, DataTypes, Model, ModelStatic} from 'sequelize';
+import {getLogger} from '../logger';
+
+const logger = getLogger('pg-based-db');
+
+interface NodeMap {
+  [key: string]: Buffer;
+}
+
+interface KVStoreAttributes {
+  key: number;
+  value: Buffer;
+}
+
+class KVStoreInstance extends Model<KVStoreAttributes> implements KVStoreAttributes {
+  key!: number;
+  value!: Buffer;
+}
+
+interface LeafLengthAttributes {
+  id: number;
+  value: number;
+}
+
+class LeafLengthInstance extends Model<LeafLengthAttributes> implements LeafLengthAttributes {
+  id!: number;
+  value!: number;
+}
+
+export class PgBasedDB {
+  private sequelize;
+  private KVStore: ModelStatic<KVStoreInstance>;
+  private LeafLength: ModelStatic<LeafLengthInstance>;
+
+  constructor(sequelize: Sequelize, wordSize = 64) {
+    this.sequelize = sequelize;
+    this.KVStore = this.initKVStoreModel(this.sequelize);
+
+    this.LeafLength = this.initLeafLengthModel(this.sequelize);
+  }
+
+  private initKVStoreModel(sequelize: Sequelize): ModelStatic<KVStoreInstance> {
+    KVStoreInstance.init(
+      {
+        key: {
+          type: DataTypes.INTEGER,
+          primaryKey: true,
+        },
+        value: {
+          type: DataTypes.BLOB,
+          allowNull: false,
+        },
+      },
+      {
+        sequelize,
+        modelName: 'KVStore',
+      }
+    );
+    return KVStoreInstance;
+  }
+
+  private initLeafLengthModel(sequelize: Sequelize): ModelStatic<LeafLengthInstance> {
+    LeafLengthInstance.init(
+      {
+        id: {
+          type: DataTypes.INTEGER,
+          primaryKey: true,
+          autoIncrement: true,
+        },
+        value: {
+          type: DataTypes.INTEGER,
+          allowNull: false,
+        },
+      },
+      {
+        sequelize,
+        modelName: 'LeafLength',
+      }
+    );
+    return LeafLengthInstance;
+  }
+
+  async connect() {
+    await this.sequelize.authenticate();
```

Comment:
This should use the same connection as the rest of the DB operations, I think we should remove this from here. Same with disconnect